### DISCUSSION TITLE
Ability to parent Pegs to Columns

### DISCUSSION THREAD

**Discussion by gab3d** [opentoonz/opentoonz#6486] [Ideas]
I suspect there is no real technical limitation for that; Pegs have the usual transformations of any other object, and can currently be parented to cameras and other Pegs.

They could be more widely used for rigging purposes if they could be placed at any level of a hierarchy, and not just at the beginning, as is the case now.

**Comment by RodneyBaker** [comment]
Off hand it seems a logical progression.
Still, I find myself slightly opposed to the idea and am not entirely sure why.

The idea of Pegs seems pretty basic.  A peg is generally a static reference point stuck to an animation table or camera stand.
Its ability to be parented to camera and other pegs relates to constraint of movement relative to the table.
It's primary purpose being for registration and alignment of things placed on the pegs.

I would think for the purpose of rigging a new construct such as a joint or null might be more appropriate.
It could of course be a 'peg' in disguise but I'd personally prefer to avoid having rigged characters with 'pegs' placed throughout.

If instead of a peg this new construct was more of a column type or subxsheet container it could then contain the things assigned to its heirarchy; other objects of all types (and sure... perhaps even pegs).

I suppose I'm trying to come up with a scenario of what might go wrong if/when Pegs were to be parented to columns.
The main thing I can think of is that we'd lose the concept of the peg and its role in animation and that peg might as well just be another column or similar construct.

Perhaps we need an option to 'convert Peg to Column' or 'treat Peg as Column'?

I suppose the underlying question is... why do we need pegs to be parented to a column?
As the answer has been suggested as relates to rigged construction it seems to me that we might not want pegs for that particular problem.

Would it be detrimental to allow pegs to parent to columns?
Not sure but I'd think they'd soon lose their role as pegs and become something other.

If we were to extrapolate this a bit further we might consider Pegs to be Control Points.
Pegs seem to be a specific type of control point (or registration point) at the edges or bounds; the ends.
What then of the control points in the middle?
Is that the gist of what we are aiming for if parenting pegs to columns?

Much food for thought but I think I will stop there before I hurt myself...  ;)


Aside:  I confirm we can parent a peg to a column by adding an intermediary such as a camera.  ;)

**Comment by RodneyBaker** [comment]
![image](https://user-images.githubusercontent.com/2576432/160259220-e4c2c422-2dbc-4502-a03f-2a8061827b04.png)

I think we need to look at this again as I am able to parent pegs to columns but... not always.
Perhaps this relates to what you suggest as being 'not just at the beginning' @gab3d?

In other words... we might just be seeing a bug that prevents parenting where it otherwise would be allowed.

**Comment by RodneyBaker** [comment]
@gab3d Try this...

Create a new temp camera to use as an intermediary to parent a peg to a column.
After setting up the parenting... delete the proxy camera(s).
This should leave the peg(s) parented to the column(s).
Save and reopen to confirm all remains intact.

**Comment by RodneyBaker** [comment]
Seems to be working for me.

![image](https://user-images.githubusercontent.com/2576432/160259410-22bb487f-926d-4b6c-81da-b61a4e8c33a4.png)

Had to put together a quick rig using this approach:
![image](https://user-images.githubusercontent.com/2576432/160260645-332ddd1c-e574-4b24-808e-013fee85f6f4.png)

**Comment by gab3d** [comment]
wow! yes, it works... 😄 
obviously it does as a corner case, not devised by programmers.
but, hey! if it works without problems, why not allowing it altogether?

**Comment by gab3d** [comment]
@RodneyBaker 
With respect to your lucubration regarding the nature of pegs, well... to be honest I just see them as nulls, as they exist in any other animation software, in one way or another, and with several different names: nulls, dummies, locators, etc.
They are basically all just a point in space with all the same transformation properties as other ordinary objects.

They are very helpful in building complex hierarchies for animatable characters/props, as they allow to easily set additional degrees of freedom or pivot points for parented elements (objects/layers/cameras/lights/whatever) inside a chain.

For sure a new object called Null could be created, but I think that already having Pegs, it'd perhaps ending to be more confusing to have two nearly identical types of objects, than just using those good-old pegs for that purpose too. 😸 

One thing that would be needed though is for them (the pegs/nulls) to be visible and selectable in the Viewer (by some gizmo/manipulator), so users could act on them more intuitively (than having to go to the schematic to select them or do it from the drop-down in the Animate tool).

**Comment by gab3d** [comment]
@RodneyBaker 
as for if it's "a bug", well... I've just tested it in Toonz 7.1 and it works exactly as we see it in OT.

![imagen](https://user-images.githubusercontent.com/18104091/160262686-d317f8ab-48bb-4260-844f-fedfb59264a5.png)

![imagen](https://user-images.githubusercontent.com/18104091/160262701-43702f3c-4981-423c-8e47-a549cdc69a9d.png)

Note: in Toonz what's different is that any Column must have a Peg before the Table (and cannot be deleted).
